src/components/input/index.tsx

```diff
@@ -26,19 +33,20 @@ export class SmoothlyInput {
 		}
 	}
 	hostData() {
-		return { class: { "has-content": this.value && this.value.length > 0 } }
+		return { class: { "has-value": this.value && this.value.length > 0 } }
 	}
 	render() {
 		return [
 			<input
 				name={this.name}
-				value={this.value}
-				type={this.type}
-				placeholder={this.placeholder}
+				value={this.typeHandler.value}
+				type={this.typeHandler.type}
+				placeholder={this.typeHandler.placeholder}
 				required={this.required}
-				autocomplete={this.autocomplete}
-				pattern={this.pattern}
-				onInput={ e => this.onInput(e as UIEvent) }></input>,
+				autocomplete={this.typeHandler.autocomplete}
+				pattern={ this.typeHandler.pattern && this.typeHandler.pattern.source }
+				onKeyDown={ e => this.typeHandler.onKeyDown(e) }
```

Comment:
Inconsistency with spaces
`{this....} and { this... }`

Reply:
So far I have always have space around expression containing operators (except .) but not around simple variables etc. I guess we should have space everywhere?

Reply:
@johnnysolsson Is it better now?

Reply:
Better